python/packages/isce3/math/offsets_polyfit.py

```diff
@@ -0,0 +1,508 @@
+
+    Returns
+    -------
+    x : ndarray
+        Solution vector or matrix satisfying `N_mat @ x = rhs`.
+    L : ndarray
+        Lower-triangular Cholesky factor of `N_mat`.
+    """
+
+    L = np.linalg.cholesky(N_mat)
+    y = np.linalg.solve(L, rhs)
+    x = np.linalg.solve(L.T, y)
+
+    return x, L
+
+def invert_from_cholesky(L):
+    """
+    Compute the inverse of a symmetric positive-definite matrix from its
+    Cholesky factorization.
+
+    Given a matrix `A` such that `A = L @ L.T`, this function computes the
+    inverse of `A` as:
+
+        A_inv = (L^-1).T @ (L^-1)
+
+    Parameters
+    ----------
+    L : (M, M) array_like
+        Lower-triangular Cholesky factor of a symmetric positive-definite
+        matrix `A`.
+
+    Returns
+    -------
+    A_inv : (M, M) ndarray
+        Inverse of the original matrix `A`.
+    """
+
+    Linv = np.linalg.inv(L)
+    return Linv.T @ Linv
+
+def polyfit_offsets(
+    data,degree=2,
+    crit_value=0.1,
+    max_iterations=50,
+    minL=None, maxL=None,
+    minP=None, maxP=None,
+    prf=None, abw=None,
+    rsr=None, rbw=None):
+
+    """
+    Robustly fit 2D polynomial models to SAR line/pixel offsets with iterative
+    outlier rejection.
+
+    The function estimates two polynomial surfaces (line and pixel directions)
+    from tie-point measurements by building a 2D polynomial design matrix up to
+    `degree`, solving with weighted normal equations, and iteratively removing
+    outliers using a w-test controlled by `crit_value`.
+
+    Parameters
+    ----------
+    data : (N, 6) array_like
+        Input tie-point array where each row is:
+        ``[id, line, pixel, dL, dP, corr_peak]``.
+        - *id* : int-like identifier.
+        - *line, pixel* : coordinates of the tie point.
+        - *dL, dP* : measured offsets in line and pixel directions.
+        - *corr_peak* : correlation peak or quality metric.
+    degree : int, optional
+        Total degree of the 2D polynomial (``1`` = affine, ``2`` = quadratic, ...).
+        Default is ``2``.
+    crit_value : float, optional
+        Critical value for the w-test used to flag outliers at each iteration.
+        Smaller values are stricter. Default is ``0.1``.
+    max_iterations : int, optional
+        Maximum number of outlier-rejection iterations. Default is ``50``.
+    minL, maxL : float, optional
+        Optional bounds on acceptable *line* offsets (units of `dL`).
+        Points with `dL` outside ``[minL, maxL]`` are removed before fitting.
+    minP, maxP : float, optional
+        Optional bounds on acceptable *pixel* offsets (units of `dP`).
+        Points with `dP` outside ``[minP, maxP]`` are removed before fitting.
+    prf, abw, rsr, rbw : float, optional
+        SAR system parameters (pulse repetition frequency, azimuth bandwidth,
+        range sampling rate, range bandwidth) used, if provided, to derive
+        prior standard deviations for weighting the line/pixel residuals.
```

Comment:
```suggestion
    prf, abw : float, optional
        SAR pulse repetition frequency and azimuth bandwidth, used if provided, to derive
        prior standard deviations for weighting the line residuals. If either is not given,
        a default prior standard deviation of 0.15/1.1 will be used for line residuals. 
    rsr, rbw : float, optional 
        SAR system range sampling rate and range bandwidth used, if provided, to derive
        prior standard deviations for weighting the pixel residuals. If either is not given,
        a default prior standard deviation of 0.1/1.1 will be used for pixel residuals. 
```
These pairs are used independently in the code, so should have independent entries in the docstring.